### ISSUE TITLE
rnnAproach assesment

### ISSUE THREAD

**Issue by kren1** [group-24/Palpitate#37]
For each configuration of the network, let the model be trained with 10 different parameters (10 iterations of the loop in rnnAproach) for 20 epochs with early stopping of 2. 

Each configuration should be tested on both user dependant and user independent data set.

Network Configurations:
- 2 layers stacked LSTM, 1 MLP at output, with dropout and Gaussian noise
- 1 layers LSTM, 1 MLP at output, with dropout and Gaussian noise
- 2 layers stacked LSTM, 1 MLP at output, with dropout and Gaussian noise, with MLP also before LSTMs
-  2 layers stacked LSTM, 1 MLP at output, with dropout and Gaussian noise with MLP between LSTMs
-  2 layers stacked LSTM, 1 MLP at output, with dropout and Gaussian noise with MLP between LSTMs and before LSTMs
- all of the above with l2 regularization
- all of the above with GRUs instead of LSTMs (if this one performs really badly, you don't have to do all of them)

So create a google docs spreadsheet. For each configuration you will have two tables to fill in, one for dependant and one for independent user data. Tables should have the following columns:
(param1, param2, param3, ... , r, rmse)

You should also copy the get_RNN_model function for each configuration, so there is no ambiguity as to what was used and can be easily reproduced. Also record the best parameters and the r and rmse on the test set.

For example the first configuration should look something like:

2 layers stacked LSTM, 1 MLP at output, with dropout and Gaussian noise - [link to get_RNN_model function]

results for dependant data:
(lstm_hidden, mlp_hidden, dropout1, dropout2, r, rmse)
(256,100,0.5,0.5, 0.63, 0.75)
.
.
.
(450,150,0.4,0.7, 0.59, 0.77)

best parameters: (256,100,0.5,0.5)
performance on test set r=0.65, rmse=0.75

results for independant data:
(lstm_hidden, mlp_hidden, dropout1, dropout2, r, rmse)
(256,100,0.5,0.5, 0.03, 1.2)
.
.
.
(450,150,0.4,0.7, 0.03, 1.2)

best parameters: (256,100,0.5,0.5)
performance on test set r=0.10, rmse=1.2

The way you change between dependant data and independent data is to change the NormalizedSubjectSplitSpectrograms class. The current versions gets the independent one. There is some commented code that can get the dependant one. The main idea is to merge train and validation set shuffle them and then split them again to get the dependant data. Same can be done with the test set but there might be no need.

**Comment by kren1** [comment]
Also you can use the cache to get this running, but you will have to rebuild it, while recording which subject_state_ids are in which set, so that the same sets can be used in baseline approach and we have a nice comparison between the data.
